### ISSUE TITLE
concurent edit protection

### ISSUE THREAD

**Issue by Cyrille37** [umap-project/umap#250]
Hi,

It should be nice to have a lock enabled when starting a map edition (like in wiki or wordpress). Now days without it we can lost work when another user loaded the map before us, then saving its work after us.

Cheers, Cyrille.

**Comment by yohanboniface** [comment]
You should have an alert in that case. Have you experienced the use case without alert?

**Comment by Cyrille37** [comment]
There is of course an alert, that bring a possible action (a link) : remove the lock and take the edit lead.
This lock remove action is necessary, as a lock timeout.
If I not clear enough, I can try in French ;-) My english is not so famous !

**Comment by VladimirSlavik** [comment]
Let me rephrase Cyrille's idea:
Currently, there is no way to know someone else is editing until you try to save - and then one of the editors must lose their work. The proposal is to introduce a "lock" of the map when editing. This lock is communicated to the server and thus also to the other users/editors. If they wish to edit the map they could see someone is editing and wait for unlocking or lock timeout.

**Comment by sylvain-m** [comment]
I am very interested in this proposal, particularly in Vladimir's rephrasing !

**Comment by Cyrille37** [comment]
Duplicate of #468
